### ISSUE TITLE
Only parse sample names from files if given congif

### ISSUE THREAD

**Issue by BJWiley233** [ewels/MultiQC#1390]
I hope this is already possible.  

I have pipeline that gives same sample name "final" to most of the files I want to use for multiqc. Instead I copy the files out and prefix them with the sample name. How can I tell multiqc to ONLY look at the file names and disregard parsing files for sample name. I created config like this

```yaml
sp:
  picard/markdups:
    fn: "*.merged.NameSorted.mark_dups_metrics.txt"
  picard/alignment_metrics:
    fn: "*.AlignmentSummaryMetrics.txt"
  picard/hsmetrics:
    fn: "*.roi-HsMetrics.txt"
  picard/insertsize:
    fn: "*.InsertSizeMetrics.txt"
  fastqc:
    fn: "*_fastqc.zip"
```

and file names:

```
CHAJ-470026-281.AlignmentSummaryMetrics.txt
CHAJ-470026-281.InsertSizeMetrics.txt
CHAJ-470026-281.VerifyBamId.depthSM
CHAJ-470026-281.VerifyBamId.selfSM
CHAJ-470026-281.bam.flagstat
CHAJ-470026-281.base-clinvar-HsMetrics.txt
CHAJ-470026-281.merged.NameSorted.mark_dups_metrics.txt
CHAJ-470026-281.roi-HsMetrics.txt
CHAJ-470026-281.target-acmg_genes-HsMetrics.txt
CHAJ-470026-281_fastqc.html
CHAJ-470026-281_fastqc.zip
```

**Comment by ewels** [comment]
Sorry, currently you can't, but it would be nice to have such an option.

Duplicate of https://github.com/ewels/MultiQC/issues/890 really (also https://github.com/ewels/MultiQC/issues/949 is similar), but I will keep this open as it more obviously general and not module-specific.

**Comment by BJWiley233** [comment]
Cool thanks.  Yes def not module specific but config.yaml specific.

**Comment by ewels** [comment]
Hi @BJWiley233,

I have added new functionality to take the log filename as the sample name as a general feature for any module / file search pattern in fa84c47f7d77284ee8fe1335132af5a6e6b4c5e2 and will be included in the MultiQC v1.11 release.

The documentation for this new feature is here: https://multiqc.info/docs/#using-log-filenames-as-sample-names

I think this should solve the issue you were getting above. If you find any problems, let me know.

Cheers,

Phil
